eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs

```diff
@@ -254,46 +187,162 @@ instance EraPParams era => NoThunks (GovernanceActionState era)
-    -- error because consensus already has golden tests for Conway
-    encCBOR (mempty :: Map () ())
-
-instance Era era => DecCBOR (ConwayTallyState era) where
-  decCBOR = do
-    _ :: Map () () <- decCBOR
-    -- FIXME: implement the actual decoder. Has to put this ugly thing here instead of an
-    -- error because consensus already has golden tests for Conway
-    pure (ConwayTallyState mempty)
-
-instance Era era => ToCBOR (ConwayTallyState era) where
-  toCBOR = toEraCBOR @era
-
-instance Era era => FromCBOR (ConwayTallyState era) where
-  fromCBOR = fromEraCBOR @era
-
-instance Crypto c => EraGovernance (ConwayEra c) where
-  type GovernanceState (ConwayEra c) = ConwayTallyState (ConwayEra c)
-  emptyGovernanceState = ConwayTallyState mempty
+data EnactState era = EnactState
+  { ensCC :: !(StrictMaybe (Set (KeyHash 'Voting (EraCrypto era)), Rational))
+  , ensConstitution :: !(SafeHash (EraCrypto era) ByteString)
+  , ensProtVer :: !ProtVer
+  , ensPParams :: !(PParams era)
+  }
+  deriving (Generic)
+
+deriving instance Eq (PParams era) => Eq (EnactState era)
+
+deriving instance Show (PParams era) => Show (EnactState era)
+
+instance (Era era, EraPParams era) => Default (EnactState era) where
+  def =
+    EnactState
+      def
+      def
+      (ProtVer (eraProtVerLow @era) 0)
+      def
+
+instance (Era era, EraPParams era) => DecCBOR (EnactState era) where
+  decCBOR =
+    decode $
+      RecD EnactState
+        <! From
+        <! From
+        <! From
+        <! From
+
+instance (Era era, EraPParams era) => EncCBOR (EnactState era) where
+  encCBOR EnactState {..} =
+    encode $
+      Rec EnactState
+        !> To ensCC
+        !> To ensConstitution
+        !> To ensProtVer
+        !> To ensPParams
+
+instance EraPParams era => NFData (EnactState era)
+
+instance EraPParams era => NoThunks (EnactState era)
+
+data RatifyState era = RatifyState
+  { rsES :: !(EnactState era)
+  , rsFuture ::
+      !( StrictSeq
+          (GovernanceActionId (EraCrypto era), GovernanceActionState era)
+       )
+  }
+  deriving (Generic, Eq, Show)
+
+instance EraPParams era => Default (RatifyState era)
+
+instance (Era era, EraPParams era) => DecCBOR (RatifyState era) where
+  decCBOR =
+    decode $
+      RecD RatifyState
+        <! From
+        <! From
+
+instance (Era era, EraPParams era) => EncCBOR (RatifyState era) where
+  encCBOR RatifyState {..} =
+    encode $
+      Rec RatifyState
+        !> To rsES
+        !> To rsFuture
+
+instance EraPParams era => NFData (RatifyState era)
+
+instance EraPParams era => NoThunks (RatifyState era)
+
+data ConwayGovernance era = ConwayGovernance
+  { cgTally :: !(ConwayTallyState era)
+  , cgRatify :: !(RatifyState era)
+  , cgVoterRoles :: !(Map (Credential 'Voting (EraCrypto era)) VoterRole)
+  }
+  deriving (Generic, Eq, Show)
+
+cgTallyL :: Lens' (ConwayGovernance era) (ConwayTallyState era)
+cgTallyL = lens cgTally (\x y -> x {cgTally = y})
```

Comment:
Why only one lens? Might as well define for the other two fields: `cgRatify` and `cgVoterRoles`.